src/Home/useLocationIsNeededStatus.ts

```diff
@@ -0,0 +1,27 @@
+import { useEffect, useState } from "react"
+import { Platform } from "react-native"
+
+import { doesDeviceSupportLocationlessScanning } from "../gaen/nativeModule"
+
+export const useLocationIsNeededStatus = (): boolean => {
+  const [isLocationNeeded, setIsLocationNeeded] = useState(false)
+
+  useEffect(() => {
+    const determineLocationIsNeeded = async () => {
+      const supportsLocationlessScanning = await determineSupportsLocationlessScanning()
+      setIsLocationNeeded(!supportsLocationlessScanning)
+    }
+
+    determineLocationIsNeeded()
+  }, [])
```

Comment:
Can someone explain why all this stuff needs to happen in the `useEffect` with the `[]` dependency array?

Reply:
Since the setIsLocationNeeded is set on a component mounting, we have to wait for it to be fetched before setting. The empty dependency array is akin to componentDidMount - meaning it won't be run again after it is mounted.

Not sure if that is what you were asking - but hope it helps.

Reply:
Ah cool I see. So it prevents this from re-running on every state change?

Reply:
From the useEffect docs:

`If you want to run an effect and clean it up only once (on mount and unmount), you can pass an empty array ([]) as a second argument. This tells React that your effect doesn’t depend on any values from props or state, so it never needs to re-run. This isn’t handled as a special case — it follows directly from how the dependencies array always works.

If you pass an empty array ([]), the props and state inside the effect will always have their initial values. While passing [] as the second argument is closer to the familiar componentDidMount and componentWillUnmount mental model, there are usually better solutions to avoid re-running effects too often. Also, don’t forget that React defers running useEffect until after the browser has painted, so doing extra work is less of a problem.`

Reply:
Thanks!